Joanna Lau Onboarding

joannalauu opened the pull request:
# Purpose
Completed the GS on-boarding task. Include a screenshot of the front-end of the application.

- [x]  I have added my name to the onboarding title (required)
- [x]  I have added a screenshot of the logs printed by the logger middleware (required)
- [x]  I am interested to work on the frontend (optional)

# New Changes
1) Implemented a function that validates whether the command has the same number of parameters and corresponding formats.

2) Implemented POST command and DELETE command endpoints, allowing the list of commands saved in the sql database to be updated. 

3) Implemented logger middleware. This logs all incoming request methods and urls, their corresponding response status codes, the datetime of the request, and the duration of execution in a log file. A screenshot of a log is attached.

<img width="1218" height="127" alt="sample log" src="https://github.com/user-attachments/assets/d4a882fc-d69a-48f1-b306-23dd6a37a602" />


# Testing
The backend passes all unit tests in the repository.

<img width="1137" height="829" alt="frontend" src="https://github.com/user-attachments/assets/371a1c86-6126-4b30-bf02-ee89035e0286" />

Commit:
onboarding submission completed - tests passed

Commit:
improved message log formating

Adityya-K commented on backend/api/endpoints/command.py:
```diff
@@ -23,23 +23,35 @@ def get_commands(db: Session = Depends(get_db)):
+        raise HTTPException(status_code=404, detail="Command not found")
+    db.delete(command)
+    db.commit()
+    query = select(Command)
```
This seems like repeated functionality, you can always use the `get_commands()` function!

Adityya-K commented on backend/api/middlewares/logger_middleware.py:
```diff
@@ -19,4 +21,8 @@ async def dispatch(
         # TODO:(Member) Finish implementing this method
         response = await call_next(request)
+        
+        logger.info(
```
Could you also log the following:
- Any response headers
- The body of the response
- The time it took for the command to execute (how long the `call_next()` function took)
- When the command was called

Adityya-K commented on backend/api/middlewares/logger_middleware.py:
```diff
@@ -1,8 +1,10 @@
 from fastapi import Request, Response
 from starlette.middleware.base import BaseHTTPMiddleware
-
+from loguru import logger
```
Remove unused imports

Adityya-K replied:
We should be using the ground station logger here, not the one for loguru!

Adityya-K reviewed (changes requested):
Good start, just a few things to change and you'll be good!!

Commit:
fixed submission errors: added more info to logs, made delete command endpoint more concise, remove unused imports

Commit:
remove commented out code

Commit:
improve logger wording

joannalauu commented:
Updated onboarding submission. Changes as follows:

1. Removed unused import in logger_middleware

2. Updated repeated functionality in DELETE command endpoint

3. Updated logger middleware to make it log more information. The following info is now logged – all information is separated by a semicolon for easier readability: 
a) Request method and url
b) Response Code
c) Response Headers
d) Response Body
e) DateTime of request
d) Execution Duration of command

<img width="1031" height="806" alt="logger ex" src="https://github.com/user-attachments/assets/2ae87fad-9adf-4beb-bacb-5a7fb0bf7ede" />

Please let me know if there is anything else you'd like me to change or add!!

Adityya-K reviewed (changes requested):
Just one more thing to resolve and I can approve your onboarding!

Commit:
used backend.utils.logging instead of loguru for logger middleware and create type check for json response body

Commit:
use response headers variable instead of retreiving again

joannalauu commented:
Changes as follows:
 
 1. Updated logger in logger_middleware to be imported from backend.utils.logging instead of loguru

2. Added error handling for a JSONDecodeError I uncovered in logger_middleware

Please let me know if this works!!

Adityya-K commented:
LGTM!
